deploy/olm-catalog/ibm-block-csi-operator/1.10.0/manifests/csi.ibm.com_hostdefinitions.yaml

```diff
@@ -0,0 +1,91 @@
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    controller-gen.kubebuilder.io/version: v0.7.0
+  creationTimestamp: null
+  labels:
+    app.kubernetes.io/instance: ibm-block-csi-operator
+    app.kubernetes.io/managed-by: ibm-block-csi-operator
+    app.kubernetes.io/name: ibm-block-csi-operator
+    csi: ibm
+    product: ibm-block-csi-driver
+    release: v1.10.0
+  name: hostdefinitions.csi.ibm.com
+spec:
+  group: csi.ibm.com
+  names:
+    kind: HostDefinition
+    listKind: HostDefinitionList
+    plural: hostdefinitions
+    singular: hostdefinition
+  scope: Cluster
+  versions:
+  - additionalPrinterColumns:
+    - jsonPath: .metadata.creationTimestamp
+      name: Age
+      type: date
+    - jsonPath: .status.phase
+      name: Phase
+      type: string
+    - jsonPath: .spec.hostDefinition.managementAddress
+      name: Storage
+      type: string
+    - jsonPath: .spec.hostDefinition.nodeName
+      name: Node
+      type: string
+    name: v1
+    schema:
+      openAPIV3Schema:
+        description: HostDefinition is the Schema for the hostdefinitions API
+        properties:
+          apiVersion:
+            description: 'APIVersion defines the versioned schema of this representation of an object. Servers should convert recognized schemas to the latest internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
+            type: string
+          kind:
+            description: 'Kind is a string value representing the REST resource this object represents. Servers may infer this from the endpoint the client submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
+            type: string
+          metadata:
+            type: object
+          spec:
+            description: HostDefinitionSpec defines the desired state of HostDefinition
+            properties:
+              hostDefinition:
```

Comment:
is it normal to have such structure inside the spec?
or is it more common to have a "flat" spec?

Reply:
this is auto-generated from [here](https://github.com/IBM/ibm-block-csi-operator/blob/0b58811d5f99ae6414c49d9461cca83bd3d61aaa/api/v1/hostdefinition_types.go#L43)